conbench/entities/benchmark_result.py

```diff
@@ -576,7 +580,31 @@ class _BenchmarkFacadeSchemaCreate(marshmallow.Schema):
     tags = marshmallow.fields.Dict(
         required=True,
         metadata={
-            "description": "Details that define the individual benchmark case that is being run (e.g. name, query type, data source, parameters). These are details about a benchmark that define different cases, for example: for a file reading benchmark, some tags might be: the data source being read, the compression that file is written in, the output format, etc."
+            "description": conbench.util.dedent_rejoin(
+                """
+                The set of key/value pairs that defines a benchmark case
+                permutation. Top-level keys must be strings. Allowed value
+                types are ... (to be specified).
+
+                The special key "name" must be provided with a string value: it
+                indicates the name of the conceptual benchmark that was
+                performed for obtaining the result at hand. All case
+                permutations of a conceptual benchmark by definition have this
+                name in common.
+
+                We advise that each unique case (as defined by the complete set
+                of key/value pairs) indeed corresponds to unique benchmarking
+                behavior. That is, typically, these key/value pairs directly
+                correspond to input parameters to your benchmarking method.
+
+                Example: a conceptual benchmark with name "foo-write-file"
+                might have meaningful case permutations involving tag names
+                such as `compression_method` (values: `gzip`, `lzma`, ...),
+                `file_format` (values: `csv`, `hdf5`, ...), `dataset_name`
+                (values: `foo`, `bar`, ...). For each conceptual benchmark, it
+                is OK to have no or many case permutations.
```

Comment:
These paragraphs are meant to be provocative! Would love to get your feedback.

Based on https://github.com/conbench/conbench/issues/787 and https://github.com/conbench/conbench/pull/875 (and other discussion threads) I think we're already converging to something.

Reply:
This makes sense to me based on what we've talked about!

Reply:
Thanks for this provocative language + pushing us to be precised.

> it is OK to have no or many case permutations.

Functionally speaking this is true (so long as you have `{name: "foo"}` you're fine[^1]. But practically speaking, there always ends up being at least one parameter and thus at least one case (if you have a parameter that has one value you have a single case, yeah?). So I wonder if we might rephrase this to be "It is ok to have few or many case permutations" and possibly also discuss that there's always at least one case (or else you have an conceptual benchmark with no actual instantiations)?

This hopefully will also nudge folks into thinking a little bit about their case permutations up front. One thing that we've struggled a bit with is as a benchmark evolves over time it usually grows more parameters. That's ok and natural, but currently conbench treats each one as if it were totally separate for history and comparison purposes, so there's a bit of friction on that front. We can't (and shouldn't) solve that here, but something to note that if we can get folks to thing "what are the parameters I might use" that's not a bad thing.


[^1]: Though I suspect that there totally might be code that doesn't handle there being no other tags in that dict and will do funny things if that's all you have.

Reply:
> at least one case

This is absolutely true.

> there's always at least one case (or else you have an conceptual benchmark with no actual instantiations)?

Yes! This must be true (as in: other thinking needs to change so that this _is_ true).

I didn't see this feedback before merging (hmm.. sorry!). Will adjust the text.

For implementation, we should totally cover that with tests, because it might be an interesting edge case: `name` is set, but no other keys.